internal/internal_decision_state_machine.go

```diff
@@ -1203,6 +1222,10 @@ func (h *commandsHelper) cancelTimer(timerID TimerID) commandStateMachine {
 
 func (h *commandsHelper) handleTimerClosed(timerID string) commandStateMachine {
 	command := h.getCommand(makeCommandID(commandTypeTimer, timerID))
+	// If, for whatever reason, we were going to send a timer cancel command, don't do that anymore
+	// since we already know the timer is resolved.
+	possibleCancelID := makeCommandID(commandTypeCancelTimer, timerID)
+	h.removeCommand(possibleCancelID)
```

Comment:
I think this should not be handled here, but instead be generalized and handled when we remove command for any closed commands.